File: ruby/openssl/ssl_client.rb
require 'socket'
require 'openssl'

# SSL Context - Client does not require cert/key if not being verified
context = OpenSSL::SSL::SSLContext.new
#context.cert = cert
#context.key = key

# TCP Client
tcp_client = TCPSocket.new 'localhost', 5000
ssl_client = OpenSSL::SSL::SSLSocket.new tcp_client, context
ssl_client.connect

ssl_client.puts "hello server!"
puts ssl_client.gets

File: python/networking/client_server/ssl/ssl_client.py
import socket

s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
s.connect(('localhost', 12345))
sslSocket = socket.ssl(s)
print repr(sslSocket.server())
print repr(sslSocket.issuer())
sslSocket.write('Hello secure socket\n')
s.close()